kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -150,6 +177,24 @@ object KinesisSink {
         .standard()
         .withRegion(region)
         .withCredentials(provider)
+        .withClientConfiguration(
+          new ClientConfiguration().withRetryPolicy(
+            new RetryPolicy(
+              new RetryCondition {
+                override def shouldRetry(
+                  originalRequest: AmazonWebServiceRequest,
+                  exception: AmazonClientException,
+                  retriesAttempted: Int
+                ): Boolean =
+                  retriesAttempted < 10
```

Comment:
> If I'm not missing anything, it means that if something cannot be inserted in SQS 10 times, we'll drop it?

Yes, we will stop retrying after 10 times, so that means drop it.

So currently in master (2.0.0) we also have 10 retries with JitterBackoff, set up with 1 second delay. 
```scala
 retry.JitterBackoff(maxRetries, 1.second)
```
The 10 retries is backed in how we determine when to stop retrying. 

In `2.0.1` we have 
```scala
new PredefinedBackoffStrategies.FullJitterBackoffStrategy(1000, 5 * 3600),
```
which means we start with 1 sec, but max backoff is limited to 5 mins (arbitrary numbers).

We have to balance somehow for how long we retry to not keep too much in memory to not kill collector I think.

> Do we expect that this will happen only when SQS is unavailable? If yes, do 10 attempts give it reasonable time to recover?

This is only a solution for AWS APIs glitches, so only when it's unavailable for short period of time. It will not help us when there is a sqs outage.

> Are there any other reasons we can foresee for a record not being able to be inserted in SQS? Does it make sense to generate a bad row here?

We validate the event earlier if it's not too big - that's one thing that comes to my mind. 
We also check if the sqs queue exists.
I don't know if there are other reasons?

Reply:
> We have to balance somehow for how long we retry to not keep too much in memory to not kill collector I think.

It seems to me that the number of minutes (5) and the maximum numbers of retries (10) should be able to set via configuration. If someone is very worried about losing events, he/she might want to give the collector a lot of memory so that it can keep events in-memory for a longer time in case of Kinesis and SQS outage.